### ISSUE TITLE
docker volume having driver cloudstor:azure not removing after system prune also

### ISSUE THREAD

**Issue by aryanPatilhm2** [docker/for-linux#923]
* [ ] This is a bug report
* [ ] This is a feature request
* [x] I searched existing issues before opening this one

I want to use the cloudstor:azure plugin in my project. So I created my azure storage account and I tried it but due to some error/issue I am not able to use that. After that I deleted my azure storage account and return to use my local volume. But while starting the app docker-compose throw the error to remove existing volume to use with local driver. I tried to remove the volumes but the volume is not removing. It thowing the error that,
```
Error response from daemon: remove app_mongovolume: VolumeDriver.Remove: error removing azure file share "data": dial tcp: lookup storageaccount.file.core.windows.net: no such host
```
I then tried every possible command for deleting the volume
```
docker volume prune
docker system prune
docker system prune -f -a
```
even I deleted metadata.db file from volumes folder in docker. But no luck.

### Expected behavior
Should remove the docker volumes using following command
```
docker volume rm app_mongovolume
```
### Actual behavior
Not removing the volume after system prune also.

### Steps to reproduce the behavior
I have created one issue regarding error with cloudstor:azure having the steps which I have performed . Below is the link,
https://github.com/docker/docker.github.io/issues/10244
Note: Before deleting volume & plugin, I have deleted the azure storage account.

**Output of `docker version`:**

```
Client:
 Version:           18.09.7
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        2d0083d
 Built:             Thu Jun 27 17:56:23 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.7
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       2d0083d
  Built:            Thu Jun 27 17:23:02 2019
  OS/Arch:          linux/amd64
  Experimental:     false

```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 38
Server Version: 18.09.7
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: active
 NodeID: e94ruuoikjqrkr542yb0ol8iq
 Is Manager: true
 ClusterID: o1tu59cv183fvwnvlf95bu0hp
 Managers: 1
 Nodes: 2
 Default Address Pool: 10.0.0.0/8  
 SubnetSize: 24
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 10
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Autolock Managers: false
 Root Rotation In Progress: false
 Node Address: 192.168.0.131
 Manager Addresses:
  192.168.0.131:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 894b81a4b802e4eb2a91d1ce216b8817763c29fb
runc version: 425e105d5a03fabd737a126ad93d62a9eeede87f
init version: fec3683
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.15.0-1067-oem
Operating System: Ubuntu 18.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.54GiB
Name: rt0719231
ID: RPYR:DRSB:DO56:H7SZ:ZAC4:QYST:GXNW:T5EY:NBOF:IDTP:XTEJ:IVTN
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Product License: Community Engine

WARNING: No swap limit support

```

**Additional details
Using Ubuntu 18.04.02 LTS
Will you please help me??
Thanks in advance..

**Comment by cpuguy83** [comment]
`prune` only remove local volumes.
It sounds like for some reason the driver is having problems looking up the hostname.

In any case, a `docker volume rm -f <volume>` should do the trick.
This forces the volume reference to be removed from memory.

**Comment by aryanPatilhm2** [comment]
Strange..!! I searched whole Friday on deleting those volumes. I completely delete all containers, images, networks volumes & overrelays. Also restarted docker & my machine but still there was the volume. Then I recreated the images & volume with different name for working on project. Today(Monday) I started the machine in morning and I saw that all my images & the volumes (with problematic cloudstor:azure) are gone. I once again restarted the machine to check if images are deleting or not on boot. But images not deleted this time.
May be completely shut down machine solved my problem. But I didn't understand about the deleting of my images & volumes.
